src/components/molecules/InternalShadowRoundButton/InternalShadowRoundButton.tsx

```diff
@@ -151,6 +159,7 @@ export const InternalShadowRoundButton = <C extends ElementType = "button">(
                 ? defaultIconColor
                 : null
           }
+          data-icon-for="button"
```

Comment:
A styling hook to target the icon from a parent styled-component. It's needed since there is no non-awful way to target hover or focus styles from a parent context without a bunch of JavaScript. It breaks the encapsulation of styles but perhaps CSS custom properties would be a better approach (though we haven't used them anywhere yet!)

Reply:
I don't mind this approach as it's clear what these are for. I can see things getting awfully muddled if we introduce css variables as people might start using them instead of the existing token system.

Reply:
🤔 possibly though the advantage of CSS custom props would be that you could open up specific properties to target from an ancestor which would carve a tightly controlled API across the component boundary.

E.g with this example there is a very clear contract to be maintained through future changes.

```html
<button>
  <img src="foo.png" data-icon-for="button" />
</button>
```

```css
img {
  filter: var(--oak-icon-color-filter, ${(props) => parseColorFilter(props.$colorFilter)});
}

button:hover {
  --oak-icon-color-filter: ${parseColorFilter('black')};
}
```

The current approach leaves specificity up to chance and you can target any property (or add arbitrary properties!)

```css
img {
  filter: ${(props) => parseColorFilter(props.$colorFilter)};
}

button:hover [data-icon-for="button"] img {
  filter: ${parseColorFilter('black')};
  background-color: orange!important;
}
```